lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js

```diff
@@ -29,10 +31,22 @@ export class LhcPeriodsOverviewModel extends Observable {
      */
     constructor() {
         super();
+
         this._pagination = new PaginationModel();
         this._pagination.observe(() => this.fetchLhcPeriods());
         this._pagination.itemsPerPageSelector$.observe(() => this.notify());
 
+        this._dataSource = new RemoteDataSource()
+            // eslint-disable-next-line no-return-assign
+            .feeds((observableDataCurrent) => this._pagination.itemsCount = observableDataCurrent.payload?.meta?.page.totalCount)
+            .feeds(new DataSourceTransfomer((remoteData) =>
+                remoteData.isSuccess()
+                    ? RemoteData.success(remoteData.payload.data.map((dataUnit) => this.flattenPeriodDataObject(dataUnit)))
+                    : remoteData))
+            .feeds(new PaginatedRemoteDataSource());
```

Comment:
For me, this is really unclear. I really need to think hard to understand what happens here.
what you have here should be :
```js
this._dataSource = new PaginatedRemoteDataSource();
// To be honest, I would even go further and do:
this._dataSource = new PaginatedRemoteDataSource(pagination); // this way, paginated remote data source can directly handle pagination
```